src/bundle/test-transformer.ts

```diff
@@ -52,3 +70,127 @@ export const setupTransformHook = (opts: { removeNonJsImports?: boolean } = {}):
 
     return revertTransformHook;
 };
+
+function shouldIgnoreImportFromStyleFile(err: Error): boolean {
+    const shouldIgnoreImport = IGNORE_STYLE_ERRORS.some(ignoreImportErr => {
+        return (err as Error).message.startsWith(ignoreImportErr);
+    });
+
+    if (!shouldIgnoreImport) {
+        return false;
+    }
+
+    const firstStackFrame = (err as Error).stack?.split("\n")[0] || "";
+    const filePath = firstStackFrame.split(":")[0];
+    const isStyleFilePath = STYLE_EXTESTION_RE.test(filePath);
+
+    return isStyleFilePath;
+}
+
+/*
+Replace esm module import with a Proxy.
+Examples
+1) `import pkg from "package"` -> `const pkg = new Proxy({}, {get: ..., apply: ...})`
+2) `import {a, b as c} from "package"` -> `const {a, c} = new Proxy({}, {get: ..., apply: ...})`
+*/
+function mockEsmModuleImport(t: typeof babel.types, path: NodePath<babel.types.ImportDeclaration>): void {
```

Comment:
Used method from here - https://babeljs.io/docs/babel-types.
In order to implement it I did use ast explorer - https://astexplorer.net/.